exporter/awsxrayexporter/awsxray.go

```diff
@@ -39,30 +44,42 @@ func NewTraceExporter(config configmodels.Exporter, logger *zap.Logger, cn connA
 	xrayClient := NewXRay(logger, awsConfig, session)
 	return exporterhelper.NewTraceExporterOld(
 		config,
-		func(ctx context.Context, td consumerdata.TraceData) (int, error) {
+		func(ctx context.Context, td consumerdata.TraceData) (totalDroppedSpans int, err error) {
 			logger.Debug("TraceExporter", typeLog, nameLog, zap.Int("#spans", len(td.Spans)))
-			droppedSpans, input := assembleRequest(td, logger)
-			logger.Debug("request: " + input.String())
-			output, err := xrayClient.PutTraceSegments(input)
-			if config.(*Config).LocalMode {
-				err = nil // test mode, ignore errors
-			}
-			logger.Debug("response: " + output.String())
-			if output != nil && output.UnprocessedTraceSegments != nil {
-				droppedSpans += len(output.UnprocessedTraceSegments)
+			totalDroppedSpans = 0
+			totalSpans := len(td.Spans)
+			for offset := 0; offset < totalSpans; offset += maxSegmentsPerPut {
+				nextOffset := offset + maxSegmentsPerPut
+				if nextOffset > totalSpans {
+					nextOffset = totalSpans
+				}
+				droppedSpans, input := assembleRequest(td.Spans[offset:nextOffset], logger)
+				totalDroppedSpans += droppedSpans
+				logger.Debug("request: " + input.String())
```

Comment:
Just to confirm: although they were already here before this PR do you really want to keep the debug logging here and below?

Reply:
I have used the debug logging several time to figure out where certain data elements are getting dropped.